### PR TITLE
#150523236 search for contributions

### PR DESCRIPTION
**Pull request by andela-uibrahim**
## What does this PR do?
This PR adds the functionality for users to search for contributions using query params to match the contribution title

## Description of Task to be completed?
Add search for contribution query
Write tests for the new feature

## How should this be manually tested?
clone the repository
checkout to the branch
run flask db upgrade
the navigate to the endpoint and include search params "http://localhost:5000/api/v1/contributions?q=contrib" 

## What are the relevant pivotal tracker stories?

#150523236

### CODE REVIEW

**Comment by Andela-ddiei** [comment]
LGTM

File: api/views/contributions.py [hunk 1]
```diff
@@ -31,11 +32,13 @@ def get(self):
         _contributions = Contribution.query.order_by(
                     Contribution.created_at.desc())
         if search_term:
-            _contributions = _contributions.filter(
-                Contribution.description.ilike('%{}%'.format(search_term)))
+            _contributions = _contributions.filter(and_(
+                Contribution.title.ilike('%{}%'.format(search_term)), Contribution.status=="approved"))
```
Comment (andela-oowonikoko) [inline api/views/contributions.py]: For confirmation, is the search supposed to return only approved contributions?

**Comment by andela-oowonikoko** [comment]
LGTM

**Comment by LarryWachira** [comment]
LGTM

File: api/views/contributions.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from uuid import uuid4
 from flask import request, url_for, jsonify, g, current_app
 from flask_restful import Resource
+from sqlalchemy import and_
```
Comment (danielwangai) [inline api/views/contributions.py]: `from sqlalchemy import and_`
imported but unused.

**Comment by danielwangai** [comment]
LGTM

**Pull request merged**